storage/innobase/trx/trx0purge.cc

```diff
@@ -669,7 +669,9 @@ fil_space_t *purge_sys_t::truncating_tablespace()
   if (space || srv_undo_tablespaces_active < 2 || !srv_undo_log_truncate)
     return space;
 
-  const ulint size= ulint(srv_max_undo_log_size >> srv_page_size_shift);
+  const uint32_t size=
+    uint32_t(std::min((1ULL << 32) - 1,
```

Comment:
Can we can use c++11 std::numeric_limits<uint32_t>::max() ?

Reply:
With a `ulonglong{}` widening cast (or an explicitly specified `std::min<ulonglong>`) we can: 53c6c823dc7cafefffdc93c79661cfb146ff8641.